models/auth/webauthn.go

```diff
@@ -43,7 +43,7 @@ type WebAuthnCredential struct {
 	Name            string
 	LowerName       string `xorm:"unique(s)"`
 	UserID          int64  `xorm:"INDEX unique(s)"`
-	CredentialID    string `xorm:"INDEX"`
+	CredentialID    string `xorm:"INDEX VARCHAR(500)"`
```

Comment:
This should no longer be necessary? Same obviously applies to the backport + all other places that are touched in this PR? All calculations done were probably correct, MariaDB was probably right this time and the comments about MariaDB probably just lead to confusion in the future? :wink:

Reply:
If you want to show that 410 works then fine: We can change it back.

Reply:
Migrated with https://github.com/go-gitea/gitea/pull/18756#issuecomment-1039711413:

~~~
select max(octet_length(credential_id)) from webauthn_credential;
+----------------------------------+
| max(octet_length(credential_id)) |
+----------------------------------+
|                              408 |
+----------------------------------+
1 row in set (0.000 sec)
~~~

the final issue that caused the length was at least unrelated to the column size. And it's plausible after all, so I think we can go back to the lower limit. Or at least remove the confusing comments and just say we have some extra space ...

Still, we wouldn't have discovered this issue if it was 500, as it's kinda a time bomb. It just silently breaks for small instances with ~ 450 credentials when it is about 500. The expected size should be 408, if it is more, we should see an error and investigate. Same in the future.